packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -116,10 +122,17 @@ export class RemoteMessageProcessor {
 
 		// Do a final unpack of runtime messages in case the message was not grouped, compressed, or chunked
 		unpackRuntimeMessage(message);
-		return {
-			messages: [message as InboundSequencedContainerRuntimeMessageOrSystemMessage],
-			batchStartCsn,
-		};
+		this.processorBatch.push(
+			message as InboundSequencedContainerRuntimeMessageOrSystemMessage,
+		);
+		if (this.batchStartCsn === undefined) {
+			const messages = [...this.processorBatch];
+			this.processorBatch.length = 0;
+			return {
+				messages,
+				batchStartCsn,
+			};
+		}
```

Comment:
Reading this over, I think it would be clarifying to `return undefined` at the end of the function, with a short comment explaining that this only returns something if we're at the end of a batch.

Also - reminder to update the doc comments for this function.